seaborn/matrix.py

```diff
@@ -168,6 +168,19 @@ def __init__(self, data, vmin, vmax, cmap, center, robust, annot, fmt,
         self.data = data
         self.plot_data = plot_data
         self.annot = annot
+
+        if isinstance(self.annot, bool) and self.annot:
+            self.annot_data = plot_data
+        elif not isinstance(self.annot, bool):
```

Comment:
I think this check will catch `None`? I may have lost track of what `self.annot` is by default at this point, though.

P.S. if you're going to touch this code again it might be cleaner to remove the `self.` since `annot` should still be a local variable at this point I think.